libs/designsystem/checkbox/src/checkbox.component.scss

```diff
@@ -125,27 +122,51 @@ $spacing-to-label: map.get(utils.$checkbox-radio-spacing, 'to-label');
   &.has-label {
     display: block;
 
-    @each $size, $height in utils.$checkbox-radio-sizes {
+    @each $size, $properties in utils.$checkbox-sizes {
+      :host-context(.checkbox-#{$size}),
       &.#{$size} {
         ion-checkbox::part(label) {
-          $vertical-padding: get-vertical-padding($height);
+          $vertical-padding: map.get($properties, vertical-padding);
 
           padding-top: $vertical-padding;
           padding-bottom: $vertical-padding;
         }
       }
     }
-  }
-}
 
-:host-context(kirby-item) {
-  z-index: utils.z('default'); // Makes whole kirby-item clickable above item-inner.
+    :host-context(.checkbox-xs),
```

Comment:
This should not be necessary, as far as I remember it was an old cookbook thing that leaked into our implementation
```suggestion
```